tests/integrations/tests/test_pydanticai.py

```
@@ -0,0 +1,706 @@
+            model = get_openai_model()
+
+            # Define tools as functions
+            def get_weather(location: str) -> str:
+                """Get the current weather for a location."""
+                return f"The weather in {location} is 72°F and sunny."
+
+            def calculate(expression: str) -> str:
+                """Perform a mathematical calculation."""
+                try:
+                    # Safe evaluation for simple expressions
+                    result = eval(expression.replace("x", "*").replace("×", "*"))
```

Ruff (0.14.5)

Use of possibly insecure function; consider using `ast.literal_eval`

(S307)